kernel/aster-nix/src/process/posix_thread/exit.rs

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use super::{futex::futex_wake, robust_list::wake_robust_futex, PosixThread, PosixThreadExt};
+use crate::{
+    prelude::*,
+    process::{do_exit_group, TermStatus},
+    thread::{thread_table, Thread, Tid},
+    util::write_val_to_user,
+};
+
+/// Exits the thread if the thread is a posix thread.
+///
+/// # Panics
+///
+/// If the thread is not a posix thread, this method will method.
+pub fn do_exit(thread: Arc<Thread>, term_status: TermStatus) -> Result<()> {
+    if thread.status().is_exited() {
+        return Ok(());
+    }
+    thread.exit();
+
+    let tid = thread.tid();
+
+    let posix_thread = thread.as_posix_thread().unwrap();
+
+    let mut clear_ctid = posix_thread.clear_child_tid().lock();
+    // If clear_ctid !=0 ,do a futex wake and write zero to the clear_ctid addr.
+    if *clear_ctid != 0 {
+        futex_wake(*clear_ctid, 1)?;
+        // FIXME: the correct write length?
+        write_val_to_user(*clear_ctid, &0u32).unwrap();
+        *clear_ctid = 0;
+    }
+    // exit the robust list: walk the robust list; mark futex words as dead and do futex wake
+    wake_robust_list(posix_thread, tid);
+
+    if tid != posix_thread.process().pid() {
+        // If the thread is not main thread. We don't remove main thread.
+        // Main thread are removed when the whole process is reaped.
+        thread_table::remove_thread(tid);
+    }
+
+    if is_main_thread(tid, posix_thread) || is_last_thread(posix_thread) {
+        // exit current process.
+        do_exit_group(term_status);
+    }
+
+    futex_wake(Arc::as_ptr(&posix_thread.process()) as Vaddr, 1)?;
+    Ok(())
+}
+
+fn is_main_thread(tid: Tid, posix_thread: &PosixThread) -> bool {
+    let process = posix_thread.process();
+    let pid = process.pid();
+    tid == pid
+}
+
+fn is_last_thread(thread: &PosixThread) -> bool {
+    let process = thread.process.upgrade().unwrap();
```

Comment:
Can these methods like `is_main_thread` and `is_last_thread` be methods of `PosixThread`? These function names sound more like they should be methods.